README.md
# AA Discord Ping Formatter

[![Version](https://img.shields.io/pypi/v/aa-discord-ping-formatter?label=release)](https://pypi.org/project/aa-discord-ping-formatter/)
[![License](https://img.shields.io/badge/license-GPLv3-green)](https://pypi.org/project/aa-discord-ping-formatter/)
[![Python](https://img.shields.io/pypi/pyversions/aa-discord-ping-formatter)](https://pypi.org/project/aa-discord-ping-formatter/)
[![Django](https://img.shields.io/pypi/djversions/aa-discord-ping-formatter?label=django)](https://pypi.org/project/aa-discord-ping-formatter/)
[![PyPI Downloads](https://img.shields.io/pypi/dm/aa-discord-ping-formatter)](https://pypi.org/project/aa-discord-ping-formatter/)
[![Code Style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](http://black.readthedocs.io/en/latest/)

App for formatting pings for Discord in Alliance Auth

## Archived

This app has been discontinued in favor of [AA Fleet Pings](https://github.com/ppfeufer/aa-fleetpings) and this Repository has been archived.

## Contents

- [Installation](#installation)
- [Updating](#updating)
- [Screenshots](#screenshots)
- [Configuration](#configuration)
- [Change Log](CHANGELOG.md)

## Installation

**Important**: This app is a plugin for Alliance Auth. If you don't have Alliance Auth running already, please install it first before proceeding. (see the official [AA installation guide](https://allianceauth.readthedocs.io/en/latest/installation/allianceauth.html) for details)

### Step 1 - Install app

Make sure you are in the virtual environment (venv) of your Alliance Auth installation. Then install the latest version:

```bash
pip install aa-discord-ping-formatter
```

### Step 2 - Update your AA settings

Configure your AA settings (`local.py`) as follows:

- Add `'discordpingformatter',` to `INSTALLED_APPS`


### Step 3 - Finalize the installation

Copy static files and run migrations

```bash
python manage.py collectstatic
```

```bash
python manage.py migrate
```

Restart your supervisor services for AA

### Step 4 - Setup permissions

Now you can setup permissions in Alliance Auth for your users. Add ``discordpingformatter | aa discord ping formatter | Can access this app`` to the states and/or groups you would like to have access.

## Updating

To update your existing installation of AA Discord Ping Formatter first enable your virtual environment.

Then run the following commands from your AA project directory (the one that contains `manage.py`).

```bash
pip install -U aa-discord-ping-formatter
```

```bash
python manage.py collectstatic
```

```bash
python manage.py migrate
```

Finally restart your AA supervisor services.

## Screenshots

### View in Alliance Auth

![AA View](https://raw.githubusercontent.com/ppfeufer/aa-discord-ping-formatter/master/discordpingformatter/docs/aa-view.jpg)

### Discord Ping

![Discord Ping Examples](https://raw.githubusercontent.com/ppfeufer/aa-discord-ping-formatter/master/discordpingformatter/docs/ping-examples.jpg)

_(Example for embedded ping (top) and non embedded ping (bottom))_

## Configuration

### Using Slack instead of Discord?

Don't worrie, I don't judge and you still can use this module. It supports also pings to Slack. Simply add the following to your `local.py`. 

```python
## AA Fleet Ping Formatter
AA_FLEETPINGFORMATTER_USE_SLACK = True
```

Although you cannot use your Auth groups as targets for pings with Slack.


CHANGELOG.md
# Change Log

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](http://keepachangelog.com/)
and this project adheres to [Semantic Versioning](http://semver.org/).

## [2.0.0a4] - 2020-09-13
### Changed
- Minimum AA version set to 2.7.4 since we use a feature that was introduced in this version. So make sure to update your Alliance Auth before testing this app.

## [2.0.0a3] - 2020-09-13
### Fixed
- Using Python 3 style `super()` without arguments
- Import order

### Removed
- template tags since they are no longer needed

## [2.0.0a2] - 2020-09-12
### Changed
- Groups for PingTargets can now be selected via drop down in admin and the respective Discord group_id is assigned automatically. **!! IMPORTANT !!** In case you already defined PingTargets in v2.0.0a1, you have to remove them before migrating.

## [2.0.0a1] - 2020-09-12
### Added
- Support for pings to Slack. If you're using Slack, add ``AA_FLEETPINGFORMATTER_USE_SLACK = True`` to your `local.py`

### Changed
- **!! breaking change !!** Settings are no longer in your `local.py`. You find them now in your admin backend. Which means. after updating to this version, you _have_ to re-do you setting in the admin backend of Auth. 

## [1.1.4] - 2020-08-14
### Fixed
- Little logic error in embedded prepings when no FC is set

## [1.1.3] - 2020-08-11
### Fixed
- Single quote issue finally fixed (#31)

## [1.1.2] - 2020-08-10
### Fixed
- Missing "Fleet" on embedded pings headline when no fleet type is selected
- Hopefully escaped possible single quotes in a characters name so they will not be transformed into their respective HTML-entities in a ping

## [1.1.1] - 2020-08-02
### Changed
- JS modularizing, creation of the link for aa-timezones moved to its own function
- Added a bit more information to the embedded ping, so on mobile phones or system notification there is a bit more than just `@everyone` or `@here` to see.

## [1.1.0] - 2020-07-17
### Added
- Configuration for pre-defined fleet comms, formup locations and doctrines. These can be configured in your `local.py` via `AA_DISCORDFORMATTER_FLEET_COMMS`, `AA_DISCORDFORMATTER_FLEET_DOCTRINES` and `AA_DISCORDFORMATTER_FLEET_FORMUP_LOCATIONS`. See [README](https://github.com/ppfeufer/aa-discord-ping-formatter#fleet-comms-formup-location-and-doctrine) for syntax.

## [1.0.0] - 2020-07-16
### Added
- Restriction for ping targets. Just to make sure not everyone can ping for Capitals or even Supers and Titans if these are configured ping targets. (#18)

### Changed
- Set time selection steps to 15 minutes instead of 60 in te datepicker
- Set Monday as the beginning of the week in the datepicker

### Fixed 
- Our Australian time travelers and everyone else who lives in the future (UTC+x) is now able to pre-ping fleets that are coming up in 2 hours Eve time, which might still be in their past local time, depending on how far in the future they live. (#19)

## [0.1.10] - 2020-07-16
### Added
- Option to embed automatic pings via webhook (#13)
- Embedded ping via webhook are color coded. Pre-defined fleet types are by default (Roam = green, Home Defense = yellow, StratOP = orange, CTA = red) and custom fleet types can be defined via settings (see [README](https://github.com/ppfeufer/aa-discord-ping-formatter#embed-webhook-pings))

### Changed
- Link to time zones conversion is now a named link

### Fixed
- Missing semicolons in JavaScript found their way back to where they belong

## [0.1.9] - 2020-07-14
### Added
- Ping creator at the end auf automatic pings via webhooks

## [0.1.8] - 2020-07-09
### Added
- Webhook group restrictions. Webhooks can now be restricted to certain groups (see [README](https://github.com/ppfeufer/aa-discord-ping-formatter#adding-ping-channels)), so not everyone who has access too this module can ping through all webhooks. Webhoos without restrictions are accessible for all with access to the module. (Thanks to Exiom for bringing this up)
- FC name is pre-filled with the users main character name, since the user is most likely te FC pinging for his own fleet.

### Changed
- Formup Time is now a proper date picker so there is a consistent date/time format throughout the pings
- Formup Time disabled by default. To enable it either check the Pre-Ping checkbox or disable the Formup NOW checkbox below the Formup Time field
- Formup Time is set to NOW by default

## [0.1.7] - 2020-07-08
### Added
- Link to timezones conversion on formup time if the [aa-timezones](https://github.com/ppfeufer/aa-timezones) module is installed

## [0.1.6] - 2020-07-04
### Fixed
- Ping for non default roles via Webhook (#9)

## [0.1.5] - 2020-06-24
### Added
- Configurable Discord webhooks to ping channels automagically

## [0.1.4] - 2020-06-18
### Changed
- Ping Type renamed to Ping Target in form

### Fixed
- typo in Additional Information

## [0.1.3] - 2020-06-15
### Added
- Configuration for additional ping targets and fleet types

## [0.1.2] - 2020-06-14
### Fixed
- sanitizing form field input 

## [0.1.1] - 2020-06-13
### Fixed
- there should always be an empty line before Additional Information ...

## [0.1.0] - 2020-06-13
### Added
- initial version


discordpingformatter/migrations/0003_webhook.py
# Generated by Django 2.2.15 on 2020-09-11 17:18

from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ("auth", "0011_update_proxy_permissions"),
        ("discordpingformatter", "0002_fleetcomm"),
    ]

    operations = [
        migrations.CreateModel(
            name="Webhook",
            fields=[
                (
                    "id",
                    models.AutoField(
                        auto_created=True,
                        primary_key=True,
                        serialize=False,
                        verbose_name="ID",
                    ),
                ),
                (
                    "type",
                    models.CharField(
                        choices=[("Discord", "Discord"), ("Slack", "Slack")],
                        default="Discord",
                        help_text="Is this a Discord or Slack webhook?",
                        max_length=7,
                    ),
                ),
                (
                    "channel",
                    models.CharField(
                        help_text="Name of the channel this webhook posts to",
                        max_length=64,
                        unique=True,
                    ),
                ),
                (
                    "url",
                    models.CharField(
                        help_text="URL of this webhook, e.g. https://discordapp.com/api/webhooks/123456/abcdef",
                        max_length=255,
                        unique=True,
                    ),
                ),
                (
                    "notes",
                    models.TextField(
                        blank=True,
                        default=None,
                        help_text="You can add notes about this webhook here if you want",
                        null=True,
                    ),
                ),
                (
                    "is_enabled",
                    models.BooleanField(
                        db_index=True,
                        default=True,
                        help_text="Whether this webhook is active or not",
                    ),
                ),
                (
                    "restricted_to_group",
                    models.ManyToManyField(
                        blank=True,
                        help_text="Restrict ping right to this webhook to one or more groups.",
                        related_name="webhook_require_groups",
                        to="auth.Group",
                    ),
                ),
            ],
            options={
                "verbose_name": "Webhook",
                "verbose_name_plural": "Webhooks",
                "default_permissions": (),
            },
        ),
    ]
